reactive-oracle-client - connection URI syntax is unclear

### Describe the bug

What is the current situation with proper URI syntax?

From previous bug reports, I'm getting a confusing message:
- https://github.com/quarkusio/quarkus/issues/23439
- https://github.com/quarkusio/quarkus/issues/23728

I did my own experiments with two formats (using `gvenzl/oracle-xe:21-slim` image):
1. `oracle:thin:@localhost:49169:quarkus_test`
    - This worked up until 2.7.5.Final.
2. `oracle:thin:@localhost:49169/quarkus_test`
    - Works since 2.7.6.Final and later.

Vert.x documentation mentions option 1.: https://vertx.io/docs/vertx-oracle-client/java/#_connection_uri.
Quarkus documentation is vague:
- https://quarkus.io/version/main/guides/reactive-sql-clients
- https://quarkus.io/version/main/guides/datasource

The new format (option 2.) is more in line with other reactive DB clients. But it feels to me that, to some extent, users have to guess the correct format.

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

> > I did my own experiments with two formats (using `gvenzl/oracle-xe:21-slim` image):
> > ```
> > 1. `oracle:thin:@localhost:49169:quarkus_test`
> >    
> >    * This worked up until 2.7.5.Final.
> > ```
> 
> It doesn't work since 2.7.6? What's the error message?

Here is what I'm getting with 2.7.6 and later:
```
10:07:38,577 INFO  [app] 10:07:36,776 Mutiny had to drop the following exception: java.sql.SQLException: Listener refused the connection with the following error:
10:07:38,578 INFO  [app] ORA-12505, TNS:listener does not currently know of SID given in connect descriptor
10:07:38,579 INFO  [app]   (CONNECTION_ID=16HA7wUdRS2i9fM1ZShLPA==)
10:07:38,580 INFO  [app] 	at oracle.jdbc.driver.T4CConnection.handleLogonNetException(T4CConnection.java:903)
10:07:38,581 INFO  [app] 	at oracle.jdbc.driver.T4CConnection.logon(T4CConnection.java:708)
10:07:38,581 INFO  [app] 	at oracle.jdbc.driver.PhysicalConnection.connect(PhysicalConnection.java:1089)
10:07:38,582 INFO  [app] 	at oracle.jdbc.driver.T4CDriverExtension.getConnection(T4CDriverExtension.java:90)
10:07:38,583 INFO  [app] 	at oracle.jdbc.driver.OracleDriver.connect(OracleDriver.java:733)
10:07:38,584 INFO  [app] 	at oracle.jdbc.datasource.impl.OracleDataSource.getPhysicalConnection(OracleDataSource.java:681)
10:07:38,585 INFO  [app] 	at oracle.jdbc.datasource.impl.OracleDataSource.getConnection(OracleDataSource.java:375)
10:07:38,586 INFO  [app] 	at oracle.jdbc.datasource.impl.OracleDataSource.getConnectionInternal(OracleDataSource.java:2120)
10:07:38,587 INFO  [app] 	at oracle.jdbc.datasource.impl.OracleDataSource$1.build(OracleDataSource.java:1845)
10:07:38,588 INFO  [app] 	at oracle.jdbc.datasource.impl.OracleDataSource$1.build(OracleDataSource.java:1779)
10:07:38,589 INFO  [app] 	at io.vertx.oracleclient.impl.OracleConnectionFactory.lambda$connect$0(OracleConnectionFactory.java:50)
10:07:38,590 INFO  [app] 	at io.vertx.core.impl.ContextImpl.lambda$null$0(ContextImpl.java:159)
10:07:38,591 INFO  [app] 	at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:100)
10:07:38,592 INFO  [app] 	at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$1(ContextImpl.java:157)
10:07:38,592 INFO  [app] 	at io.vertx.core.impl.TaskQueue.run(TaskQueue.java:76)
10:07:38,592 INFO  [app] 	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:559)
10:07:38,593 INFO  [app] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
10:07:38,593 INFO  [app] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
10:07:38,594 INFO  [app] 	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
10:07:38,594 INFO  [app] 	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
10:07:38,595 INFO  [app] 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
10:07:38,595 INFO  [app] 	at java.base/java.lang.Thread.run(Thread.java:829)
10:07:38,596 INFO  [app] Caused by: oracle.net.ns.NetException: Listener refused the connection with the following error:
10:07:38,596 INFO  [app] ORA-12505, TNS:listener does not currently know of SID given in connect descriptor
10:07:38,596 INFO  [app]   (CONNECTION_ID=16HA7wUdRS2i9fM1ZShLPA==)
10:07:38,596 INFO  [app] 	at oracle.net.ns.NSProtocolNIO.createRefusePacketException(NSProtocolNIO.java:816)
10:07:38,597 INFO  [app] 	at oracle.net.ns.NSProtocolNIO.handleConnectPacketResponse(NSProtocolNIO.java:396)
10:07:38,597 INFO  [app] 	at oracle.net.ns.NSProtocolNIO.negotiateConnection(NSProtocolNIO.java:207)
10:07:38,598 INFO  [app] 	at oracle.net.ns.NSProtocol.connect(NSProtocol.java:350)
10:07:38,598 INFO  [app] 	at oracle.jdbc.driver.T4CConnection.connect(T4CConnection.java:2558)
10:07:38,598 INFO  [app] 	at oracle.jdbc.driver.T4CConnection.logon(T4CConnection.java:667)
10:07:38,599 INFO  [app] 	... 20 more
```

> > Vert.x documentation mentions option 1.: https://vertx.io/docs/vertx-oracle-client/java/#_connection_uri. Quarkus documentation is vague:
> > ```
> > * https://quarkus.io/version/main/guides/reactive-sql-clients
> > 
> > * https://quarkus.io/version/main/guides/datasource
> > ```
> 
> This example should be updated. Would you mind filing an issue on `vertx-sql-client` repo ?

Sure, here it is: https://github.com/eclipse-vertx/vertx-sql-client/issues/1209.

--- a/docs/src/main/asciidoc/datasource.adoc
+++ b/docs/src/main/asciidoc/datasource.adoc
@@ -649,3 +649,93 @@ include::{generated-dir}/config/quarkus-reactive-oracle-client.adoc[opts=optiona
 === Reactive PostgreSQL Specific Configuration
 
 include::{generated-dir}/config/quarkus-reactive-pg-client.adoc[opts=optional, leveloffset=+1]
+
+
+[[reactive-url]]
+== Reactive Datasource URL Reference
+
+=== DB2
+
+`db2://[user[:[password]]@]host[:port][/database][?<key1>=<value1>[&<key2>=<value2>]]`
+
+Example:: `db2://dbuser:secretpassword@database.server.com:50000/mydb`
+
+Currently, the client supports the following parameter keys:
+
+* `host`
+* `port`
+* `user`
+* `password`
+* `database`
+
+NOTE: Configuring parameters in connection URL will override the default properties.
+
+=== Microsoft SQL Server
+
+`sqlserver://[user[:[password]]@]host[:port][/database][?<key1>=<value1>[&<key2>=<value2>]]`
+
+Example:: `sqlserver://dbuser:secretpassword@database.server.com:1433/mydb`
+
+Currently, the client supports the following parameter keys:
+
+* `host`
+* `port`
+* `user`
+* `password`
+* `database`
+
+NOTE: Configuring parameters in connection URL will override the default properties.
+
+=== MySQL / MariaDB
+
+`mysql://[user[:[password]]@]host[:port][/database][?<key1>=<value1>[&<key2>=<value2>]]`
+
+Example:: `mysql://dbuser:secretpassword@database.server.com:3211/mydb`
+
+Currently, the client supports the following parameter keys (case-insensitive):
+
+* `host`
+* `port`
+* `user`
+* `password`
+* `schema`
+* `socket`
+* `useAffectedRows`
+
+NOTE: Configuring parameters in connection URL will override the default properties.
+
+=== Oracle
+
+==== EZConnect Format
+
+`oracle:thin:@[[protocol:]//]host[:port][/service_name][:server_mode][/instance_name][?connection properties]`
+
+Example:: `oracle:thin:@mydbhost1:5521/mydbservice?connect_timeout=10sec`
+
+==== TNS Alias Format
+
+`oracle:thin:@<alias_name>[?connection properties]`
+
+Example:: `oracle:thin:@prod_db?TNS_ADMIN=/work/tns/`
+
+=== PostgreSQL
+
+`postgresql://[user[:[password]]@]host[:port][/database][?<key1>=<value1>[&<key2>=<value2>]]`
+
+Example:: `postgresql://dbuser:secretpassword@database.server.com:5432/mydb`
+
+Currently, the client supports the following parameter keys:
+
+* `host`
+* `port`
+* `user`
+* `password`
+* `dbname`
+* `sslmode`
+* additional properties, including:
+** `application_name`
+** `fallback_application_name`
+** `search_path`
+** `options`
+
+NOTE: Configuring parameters in connection URL will override the default properties.

